server/py/services/api/crud/alert_activation.py

```diff
@@ -13,10 +13,62 @@
 # limitations under the License.
 #
 
+import sqlalchemy.orm
+
+import mlrun.common.schemas.alert
 import mlrun.utils.singleton
 
+import framework.utils.singletons.db
+
 
 class AlertActivation(
     metaclass=mlrun.utils.singleton.Singleton,
 ):
-    pass
+    def store_alert_activation(
+        self,
+        session: sqlalchemy.orm.Session,
+        alert_data: mlrun.common.schemas.AlertConfig,
+        event_data: mlrun.common.schemas.Event,
+    ):
+        notifications_states = self._prepare_notifications_states(alert_data)
+
+        framework.utils.singletons.db.get_db().store_alert_activation(
+            session, alert_data, event_data, notifications_states
+        )
+
+    def list_alerts_activations(
+        self,
+        session: sqlalchemy.orm.Session,
+        project: str = "",
+    ) -> list[mlrun.common.schemas.AlertActivation]:
+        # TODO: add filters
+        project = project or mlrun.mlconf.default_project
+        return framework.utils.singletons.db.get_db().list_alerts_activations(
+            session, project
+        )
+
+    @staticmethod
+    def _prepare_notifications_states(
+        alert: mlrun.common.schemas.AlertConfig,
+    ) -> list[mlrun.common.schemas.NotificationState]:
+        # process the notifications associated with the provided alert and construct a list of NotificationState objects
+        # each NotificationState represents a unique type of notification (e.g., "slack", "git") and its status.
+        # if at least one notification of that type failed, the error message from the most recent failure will
+        # be stored. if no failure occurred, the status will be an empty string indicating success.
+        notification_states = {}
+
+        for alert_notification in alert.notifications:
+            notification_kind = alert_notification.notification.kind
+            notification_reason = alert_notification.notification.reason
+
+            notification_states.setdefault(
+                notification_kind,
+                notification_reason if notification_reason is not None else "",
+            )
+
+        notification_states = [
+            mlrun.common.schemas.NotificationState(kind=kind, err=err)
+            for kind, err in notification_states.items()
+        ]
+
+        return notification_states
```

Comment:
```suggestion
    notification_states =   [
        mlrun.common.schemas.NotificationState(
            kind=alert_notification.notification.kind,
            err=alert_notification.notification.reason,
        )
        for alert_notification in alert.notifications
    ]
    return notification_states
```

Reply:
this way for each notification we will have a different entry in the list, which is not aligned with the requirements that says we should have only one entry for each notification type..

Reply:
What about this ^, why did you eventually change it to the full list instead of unique item per kind?